src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -71,16 +72,24 @@ def invalid_metric_tags(tags: Mapping[str, str]) -> Sequence[str]:
     return invalid_strs
 
 
+def extract_use_case_id(mri: str) -> Optional[str]:
+    use_case_mapping = {"transactions": "performance", "sessions": "release-health"}
+    if (matched := MRI_RE_PATTERN.match(mri)) is not None:
```

Comment:
Why do we need to put the use case id into the string then parse it back out again with this regex? It seems a bit fragile and I assume is not ideal for performance.. is there a way to avoid this step?

Reply:
It's definitely not ideal, the cleanest way of doing this is to have relay send us the MRI name, that way, we don't even need an intermediate type (`parsed_message`). But as we're prototyping right now, we are not sure if we will keep using the MRI name as use case ID down the line.

Reply:
Relay currently does send the MRI via the `name` field in a metric message, and this MRI contains the namespace, which we are mapping to use_case_id. 

Relay does not add a `use_case_id` field to messages, and we are moving forward with this for now, without changing how Relay formats payload and sends it to Sentry. So, yes, because of this, we have the intermediate type `ParsedMessage`, added in this PR. 

Ultimately, we could revisit this decision if needed. But for quick validation now, this is an easy way to verify the end to end and test out this platform with a new use case.

Reply:
My bad, meant to say namespace instead of name, edited my comment above